inbc-program/inbc/parser/source_app_parser.py

```diff
@@ -35,8 +35,8 @@ def application_add(args: argparse.Namespace) -> str:
 
     manifest += '<repo><repos>'
 
-    for source in args.sources:
-        manifest += '<source_pkg>' + source + '</source_pkg>'
+    source_tags = [f'<source_pkg>{source}</source_pkg>' for source in args.sources]
+    manifest += ''.join(source_tags)
```

Comment:
**Suggestion:** Ensure that the 'args.sources' list is not empty before creating 'source_tags' to avoid generating an empty 'repos' XML element. [possible issue]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    manifest += '<repo><repos>'
    if args.sources:
        source_tags = [f'<source_pkg>{source}</source_pkg>' for source in args.sources]
        manifest += ''.join(source_tags)
```